Docs: Relabel Automatic Redirects as "Incoming links: Best practices and redirects" (Diátaxis) (#9896)
readthedocs/readthedocs.org: docs/user/user-defined-redirects.rst

the bad user experience of a 404 page is usually best to avoid.

`Built-in redirects`_ ⬇️
    Allows for simple and long-term sharing of external references to your documentation.

`User-defined redirects`_ ⬇️
    Makes it easier to move contents around (see: )

.. seealso::

   :doc:`/guides/redirects`
     This guide shows you how to add redirects with practical examples.
   :doc:`/automatic-redirects`
     Information and tips about creating and handling external references.
   :doc:`/guides/deprecating-content`
     A guide to deprecating features and other topics in a documentation.

Special features
----------------

- By default, redirects are followed only if the requested page doesn't exist
  (*404 File Not Found* error).
  If you need to apply a redirect for files that exist,
  **This option is only available on some plan levels**.
  Please ask support if you need it for some reason.
- :ref:`user-defined-redirects:page redirects` and :ref:`user-defined-redirects:exact redirects`
  can redirect to URLs outside Read the Docs,
  just include the protocol in ``To URL``, e.g ``https://example.com``.

Built-in redirects
------------------

> This article explains the redirects that are automatically active for all Read the Docs projects and how they are useful.
> Automatic redirects on Read the Docs are supported long-term.
> 
> Built-in redirects are especially useful for creating and sharing **incoming links**
> which is discussed indepth in :doc:`/automatic-redirects`.

Reviewer: ```suggestion
This section explains the redirects that are automatically active for all projects and how they are useful.
Built-in redirects are especially useful for creating and sharing incoming links,
which is discussed indepth in :doc:`/automatic-redirects`.
```

Page redirects at ``/page/``
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

You can link to a specific page and have it redirect to your default version,
allowing you to create links on external sources that are always up to date.
This is done with the ``/page/`` URL prefix.

For instance, you can reach the page you are reading now by going to https://docs.readthedocs.io/page/automatic-redirects.html.

Another way to handle this is the ``latest`` version.
You can set your ``latest`` version to a specific version and just always link to ``latest``.
You can reach this page by going to https://docs.readthedocs.io/en/latest/automatic-redirects.html.

Root URL redirect at ``/``
~~~~~~~~~~~~~~~~~~~~~~~~~~

A link to the root of your documentation (`<slug>.readthedocs.io/`) will redirect to the  :term:`default version`,
as set in your project settings.

This works for both readthedocs.io (|org_brand|), readthedocs-hosted.com (|com_brand|), and :doc:`custom domains </custom-domains>`.

For example::

>     docs.readthedocs.io -> docs.readthedocs.io/en/latest/
>     www.pip-installer.org -> www.pip-installer.org/en/latest/

Reviewer: ```suggestion
    docs.readthedocs.io -> docs.readthedocs.io/en/stable/
```

.. warning::

>    This only works for the root URL, not for internal pages.

Reviewer: What are internal pages?

Author: I don't know + this whole warning box is weird. Look at the warning text below.

Gonna delete it...

Author: I think we should combine the two warnings to a single `note::` that says

> You cannot use the root redirect to reference specific pages. `/` redirects to the default version, but `/some/page.html` will not redirect to `/en/latest/some/page.html`. Use the `/page/` redirect rule for this purpose.

Author: Made it a warning box.

   It's designed to redirect people from `/` to the default version of your documentation,
   since serving up a 404 here would be bad user experience.

> .. note::
>    If the "develop" branch was designated as the default version,
>    then ``http://pip.readthedocs.io/`` would redirect to ``http://pip.readthedocs.io/en/develop``.
>    But, it's not a universal redirecting solution.
>    So, for example, a link to an internal page like
>    ``http://pip.readthedocs.io/usage.html`` doesn't redirect to ``http://pip.readthedocs.io/en/latest/usage.html``.
> 
>    The reasoning behind this is that RTD organizes the URLs for docs so that multiple translations and multiple versions of your docs can be organized logically and consistently for all projects that RTD hosts.
>    For the way that RTD views docs,
>    ``http://pip.readthedocs.io/en/latest/`` is the root directory for your default documentation in English, not ``http://pip.readthedocs.io/``.
>    Just like ``http://pip.readthedocs.io/en/develop/`` is the root for your development documentation in English.
> 

Reviewer: ```suggestion
```

Seems this both of these notes say the same thing? I don't like using examples other people control, so we should just remove it.

> Among all the multiple versions of docs,
> you can choose which is the "default" version for Read the Docs to display,
> which usually corresponds to the git branch of the most recent official release from your project.
> 
> Shortlink with ``https://*.rtfd.io``
> ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
> 
> Links to ``rtfd.io`` are treated the same way as ``readthedocs.io``.
> They redirect the root URL to the default version of the project.
> They are intended to be easy and short for people to type.
> 
> You can reach these docs at https://docs.rtfd.io.

Reviewer: ```suggestion
You can choose which is the "default" version for Read the Docs to display.
This usually corresponds to version of the most recent official release from your project.

Shortlink with ``https://<slug>.rtfd.io``
~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~

Links to ``rtfd.io`` are treated the same way as ``readthedocs.io``.
They are intended to be easy and short for people to type.

You can reach these docs at https://docs.rtfd.io.
```

Author: Thanks! Modified your suggestion a tiny bit :+1:
